### PR TITLE
Add autokey application

### PR DESCRIPTION
**Pull request by JohnJohnstone**
###### Motivation for this change

To add the text expander autokey

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -598,6 +598,8 @@ with pkgs;
 
   autorevision = callPackage ../tools/misc/autorevision { };
 
+  autokey = callPackage ../tools/x11/autokey { };
```
Comment (hedning) [inline pkgs/top-level/all-packages.nix]: Looks like it should be `X11` not `x11`.

File: pkgs/tools/X11/autokey/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/autokey/autokey/archive/v${version}.tar.gz";
+    sha256 = "619c5b93c015ab79176f98d6a0a17f14706cf23e9b79d5ffc5c0295b2ec88612";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ xlib pyinotify dbus-python ];
+
+  pythonPath = with python3Packages;
+                  [ dbus-python ];
```
Comment (adisbladis) [inline pkgs/tools/X11/autokey/default.nix]: It's probably sufficient to only add `dbus-python` in `propagatedBuildInputs`.

File: pkgs/tools/X11/autokey/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "autokey";
+  version  = "0.94.0";
+  name = "${pname}-${version}";
```
Comment (adisbladis) [inline pkgs/tools/X11/autokey/default.nix]: When using `buildPythonApplication` and `buildPythonPackage` the name attribute defaults to  `"${pname}-${version}"' so this live can be removed.

File: pkgs/tools/X11/autokey/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchurl {
```
Comment (adisbladis) [inline pkgs/tools/X11/autokey/default.nix]: You should use `fetchPypi` or `fetchFromGitHub`.

File: pkgs/tools/X11/autokey/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    url = "https://github.com/autokey/autokey/archive/v${version}.tar.gz";
+    sha256 = "619c5b93c015ab79176f98d6a0a17f14706cf23e9b79d5ffc5c0295b2ec88612";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ xlib pyinotify dbus-python ];
+
+  pythonPath = with python3Packages;
+                  [ dbus-python ];
+
+  meta = with stdenv.lib; {
```
Comment (adisbladis) [inline pkgs/tools/X11/autokey/default.nix]: Are you willing to maintain this package? 
If so please add yourself to `lib/maintainers.nix` and reference yourself in `meta.maintainers`.

File: pkgs/tools/X11/autokey/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl, python3Packages }:
```
Comment (adisbladis) [inline pkgs/tools/X11/autokey/default.nix]: Could you please replace `callPackage` in `all-packages.nix` with `python3Packages.callPackage` and explicitly declare all inputs to the derivation in the function arguments? 

With that I mean replace the line `{ stdenv, fetchurl, python3Packages }:` with `{ stdenv, fetchurl, xlib, pyinotify, dbus-python, buildPythonApplication }:`.

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/tools/X11/autokey/default.nix (hunk 6 above)
Comment (Johnstone-Tech) [reply pkgs/tools/X11/autokey/default.nix]: Thank you the package starts building but fails due to dbus-python not being found, which is why i had dbus-python in buildinputs but no sucess

File: pkgs/tools/X11/autokey/default.nix (hunk 5 above)
Comment (Johnstone-Tech) [reply pkgs/tools/X11/autokey/default.nix]: Once i get my head round nix expressions and packaging i would gladly maintain this package as i use it daily.

**Comment by JohnJohnstone** [comment]
> close due to inaction?

Yes im not using nixpkgs anymore.

**Pull request closed without merging**